neuro_data/static_images/dataset_config.py

```diff
@@ -0,0 +1,533 @@
+                        "scan_idx",
+                        "layer",
+                        "unit_id",
+                        as_dict=True,
+                    )
+                )
+            )
+            unit_df = pd.DataFrame(unit_keys)
+            unit_df = unit_df.merge(unit_layer_df, how="left")
+            assert unit_df.layer.notnull().all(), "Missing layer for some units!"
+            return unit_df.layer.values
+
+    class Constant(dj.Part):
+        definition = """
+        -> master
+        ---
+        layer         :varchar(256)        # layer name
+        """
+        content = [
+            {"layer": "L2/3"},
+        ]
+
+        def layer(self, unit_keys):
+            return np.full(len(unit_keys), self.fetch1("layer"))
+
+
+@schema
+class AreaConfig(ConfigBase, dj.Lookup):
+    _config_type = "area"
+
+    def area(self, scan_key, unit_ids):
+        self.part_table().area(scan_key, unit_ids)
+
+    class AreaMembership(dj.Part):
+        definition = """
+        -> master
+        ---
+        """
+        content = [
+            {},
+        ]
+
+        def area(self, unit_keys):
+            unit_area_df = pd.DataFrame(
+                (
+                    (anatomy.AreaMembership & unit_keys).fetch(
+                        "animal_id",
+                        "session",
+                        "scan_idx",
+                        "brain_area",
+                        "unit_id",
+                        as_dict=True,
+                    )
+                )
+            )
+            unit_df = pd.DataFrame(unit_keys)
+            unit_df = unit_df.merge(unit_area_df, how="left")
+            assert unit_df.area.notnull().all(), "Missing area for some units!"
+            return unit_df.area.values
+
+    class Constant(dj.Part):
+        definition = """
+        -> master
+        ---
+        brain_area         :varchar(256)        # brain area name
+        """
+        content = [
+            {"brain_area": "V1"},
+        ]
+
+        def area(self, unit_keys):
+            return np.full(len(unit_keys), self.fetch1("brain_area"))
+
+
+@schema
+class StatsConfig(ConfigBase, dj.Lookup):
+    _config_type = "stats"
+
+    class NoBehFullFieldFrame(dj.Part):
+        # implemented only for full field stimulus with stimulus_type=='stimulus.Frame'
+        definition = """
+        # mimic the behavior of InputResponse with the corresponding preprocess_id to compute the statistics of input and responses, do not compute stats for behavior variables
+        -> master
+        ---
+        stats_tier="train"                 : enum("train", "test", "validation", "all")                # tier used for computing stats
+        stats_per_input                    : tinyint                                                  # whether to compute stats per input
+        linear_monitor                     : tinyint                                                  # whether to compute linear monitor
```

Comment:
I think you are referring to the 'gamma' secondary key in preprocessing, not 'linear_monitor'. 'gamma' is the indicates whether to convert inputs to the luminance space. Linearized monitor is a different thing. We can include both.

Reply:
After a second thought, I think it is better to include those parameters in InputConfig instead.